src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10895,14 +10941,35 @@ private ExpressionSyntax ParsePostFixExpression(ExpressionSyntax expr)
             }
         }
 
-        private static bool CanStartConsequenceExpression(SyntaxKind kind)
+        private bool CanStartConsequenceExpression()
         {
-            return kind is SyntaxKind.DotToken or SyntaxKind.OpenBracketToken;
+            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.QuestionToken);
+            var nextToken = this.PeekToken(1);
```

Comment:
nit: the only use for `nextToken` is to grab its `Kind`. Consider using a local for kind instead.

Reply:
Done.